game-app/game-core/src/main/java/games/strategy/triplea/delegate/data/CasualtyDetails.java

```diff
@@ -76,13 +78,41 @@ private CasualtyDetails ensureUnitsAreTakenFirst(
             getKilled().stream()
                 .filter(matcher)
                 .collect(Collectors.groupingBy(UnitOwner::new, Collectors.toList())));
-    final List<Unit> damaged =
-        ensureUnitsAreTakenFirst(
-            comparator,
-            targetsGroupedByOwnerAndType,
-            getDamaged().stream()
-                .filter(matcher)
-                .collect(Collectors.groupingBy(UnitOwner::new, Collectors.toList())));
+
+    final List<Unit> damaged = new ArrayList<>();
+
+    final Map<UnitOwner, List<Unit>> oldTargetAirUnitsToTakeHits =
```

Comment:
It's odd to know that we are dealing with only air units. Are we actually only dealing with air units here?

Reply:
Your are right, I will fix the variable name.